src/Features/Core/Portable/RemoveUnusedVariable/AbstractRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -58,12 +60,38 @@ protected override Task FixAllAsync(Document document, ImmutableArray<Diagnostic
                 else
                 {
                     var variableDeclarator = token.GetAncestor<TVariableDeclarator>();
-
                     var variableDeclarators = token.GetAncestor<TVariableDeclaration>().ChildNodes().Where(x => x is TVariableDeclarator);
 
                     if (variableDeclarators.Count() == 1)
                     {
-                        editor.RemoveNode(token.GetAncestor<TLocalDeclarationStatement>());
+                        var localDeclaration = token.GetAncestor<TLocalDeclarationStatement>();
+                        var removeOptions = SyntaxGenerator.DefaultRemoveOptions;
+
+                        if (localDeclaration.GetLeadingTrivia().Contains(t => t.IsDirective))
+                        {
+                            removeOptions |= SyntaxRemoveOptions.KeepLeadingTrivia;
+                        }
+                        else
+                        {
+                            var statementParent = localDeclaration.Parent;
+                            if (syntaxFacts.IsExecutableBlock(statementParent))
+                            {
+                                var siblings = syntaxFacts.GetExecutableBlockStatements(statementParent);
+                                var localDeclarationIndex = siblings.IndexOf(localDeclaration);
+                                if (localDeclarationIndex != 0)
+                                {
+                                    // if we're removing hte first statement in a block, then we
+                                    // want to have the elastic marker on it so that the next statement
+                                    // properly formats with the space left behind.  But if it's
+                                    // not the first statement then just keep the trivia as is
+                                    // so that the statement before and after it stay appropriately
+                                    // spaced apart.
+                                    removeOptions &= ~SyntaxRemoveOptions.AddElasticMarker;
```

Comment:
I think there are a couple not-so-important (IMO) cases this misses, but I'm mentioning them in case this sparks some related-but-more-important scenarios

```
{
  Console.Writeline();

  int unused;
}
```

Will removing `unused` will leave the blank line? If not, a test would be nice.

```
{
  Console.Writeline();

  int unused;

  Console.Writeline();
}
```

Will removing `unused` will end up with two blank lines separating the remaining statements, which _probably_ isn't what's wanted?